src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebResponseObject.CoreClr.cs

```diff
@@ -31,23 +31,28 @@ public partial class WebResponseObject
         {
             get
             {
-                var headers = new Dictionary<string, IEnumerable<string>>(StringComparer.OrdinalIgnoreCase);
-                foreach (var entry in BaseResponse.Headers)
+                if(headers == null)
                 {
-                    headers[entry.Key] = entry.Value;
-                }
-                if (BaseResponse.Content != null)
-                {
-                    foreach (var entry in BaseResponse.Content.Headers)
+                    headers = new Dictionary<string, IEnumerable<string>>(StringComparer.OrdinalIgnoreCase);
+                    foreach (var entry in BaseResponse.Headers)
                     {
                         headers[entry.Key] = entry.Value;
                     }
+                    if (BaseResponse.Content != null)
+                    {
+                        foreach (var entry in BaseResponse.Content.Headers)
+                        {
+                            headers[entry.Key] = entry.Value;
+                        }
+                    }
                 }
 
                 return headers;
             }
         }
 
+        private Dictionary<string, IEnumerable<string>> headers = null;
```

Comment:
Should it be `_headers`? [naming-conventions](https://github.com/PowerShell/PowerShell/blob/ede580e7d19c1aecdb708ba9a09a7beaee5a0a14/docs/dev-process/coding-guidelines.md#naming-conventions)

Reply:
Fixed, thanks.